ptest/cases/adcs_checkout_case.py

```
@@ -0,0 +1,202 @@
+        self.print_header("Begin ADCS Checkout Case")
+
+        # Needed so that ADCSMonitor updates its values
+        self.cycle()
+
+        self.print_rs("adcs_monitor.functional")
+        if not self.rs("adcs_monitor.functional"):
+            raise TestCaseFailure(f"ADCSC Not Functional")
+
+        self.ws("pan.state", self.mission_states.get_by_name("manual"))
+        self.ws("adcs.state", self.adcs_states.get_by_name("point_manual"))
+        self.ws("adcs_cmd.rwa_mode", self.rwa_modes.get_by_name("RWA_SPEED_CTRL"))
+        self.ws("adcs_cmd.rwa_speed_cmd", [0,0,0])
+        self.ws("dcdc.ADCSMotor_cmd", True)
+    
+        self.print_header("Finished Initialization")
+
+    def havt_checkout(self):
+        # reset all devices in case last ptest case left an "unclean state"
+        for x in range(self.havt_length):
+            self.ws(f"adcs_cmd.havt_reset{x}", True)
+
+        # adcs_controller should have applied commands.
+        self.cycle()
+
+        self.print_havt_read()
+
+        for x in range(self.havt_length):
+            if not self.havt_read[x]:
+                self.logger.put(f"Device #{x}, {self.havt_devices.get_by_num(x)} is not functional")
+
+        self.logger.put("Initial HAVT Table:")
+        self.print_havt_read()
+
+        # Note IMUGYR on left
+        # Just FC and ADCSC
+        barebones_hitl = "000111011100" + "000000"
+
+        # FC + ADCSC + MAG1 + MAG2 + GYR
+        minimal_hitl   = "111111011100" + "000000"
+            
+        # I forgot what it actually is, will update on next PR, after testing with EDU SAT
+        edu_sat        = "110111011100" + "000000"
+
+        test_beds = {barebones_hitl:"BAREBONES HITL", minimal_hitl:"MINIMAL HITL", edu_sat:"EDU SAT"}
+
+        binary_string_havt_read = ''.join(["1" if x else "0" for x in self.havt_read])
+
+        if(binary_string_havt_read not in test_beds):
+            self.print_header("UN-RECOGNIZED HITL TEST BED. MAKE SURE HAVT_READ IS AS EXPECTED!")
```

Shouldn't this cause a test case failure?

---

Wouldn't we want this to be something more like this:

    for testbed in testbeds:
        if (detected ^^ testbed) && testbed:
            testbed isn't supported

---

My logic is that if we suddenly have a new testbed, we might want to just run this script anyway to run checks on it. 

Or, if we know a bed has problems, this would just confirm those problems. Then we can still test whatever sensors/stuff is still functional.

---

This shouldn't serve as a "development" ptest case, I think it should be a hard check on functionality. If a testbed suddenly has problems, I feel like the test case should fail hard. Everything else that's a soft assert I agree with because sensor values can fluctuate unpredictably. Unrecognized testbeds, though, indicate undefined behavior.